subiquity/models/tests/test_filesystem.py

```diff
@@ -1279,6 +1279,25 @@ def test_render_includes_unmounted_new_partition(self):
         self.assertTrue(disk2.id in rendered_ids)
         self.assertTrue(disk2p1.id in rendered_ids)
 
+    def test_bind_mount(self):
+        model = make_model()
+        make_disk(model, path="/dev/vda")
+        fake_up_blockdata(model)
+        model.apply_autoinstall_config(
+            [
+               {
+                    "id": "tmpfs1",
+                    "type": "mount",
+                    "spec": "none",
+                    "path": "/tmp",
+                    "size": "4194304",
+                    "fstype": "tmpfs",
+                },
+            ]
+        )
+        rendered_mount = model.all_mounts()
+        self.assertEqual(rendered_mount[0].fstype, "tmpfs")
+
```

Comment:
Try adding something like this:
```suggestion
self.assertEqual(model.render()["storage"]["config"][0]["fstype"], "tmpfs")
```

Reply:
Hi @dbungert 

> Try adding something like this:

>  Suggested change
> self.assertEqual(model.render()["storage"]["config"][0]["fstype"], "tmpfs")

With that code  I could reproduce the issue, enabling me to debug further.

> I have a suggested addition to the test that illustrates the problem without a curtin invocation - the serialization code seems to not like the property I think? 

Indeed, the serialization code skips if the attribute starts with `_`, which is true in this specific case, and the attribute is not added to the serialization list.

> If that's true, maybe it makes more sense to abandon fstype as a property [...]
I'm not sure I fully understand what you mean by this.
If we abandoned fstype as property, then should we revert back the change to `fstype: Optional[str] = None` ?

> and fix the other code that handles the mountlikes to call to a new method that the serialization code is unaware of.

2. IIUC, I believe you mean [this code](https://github.com/canonical/subiquity/blob/b2b6e9ffef0028f72ded1d780015e424d21569d7/subiquity/models/filesystem.py#L2013) ? And append the `fstype` property to `r` ?

Reply:
> IIUC, I believe you mean [this code](https://github.com/canonical/subiquity/blob/b2b6e9ffef0028f72ded1d780015e424d21569d7/subiquity/models/filesystem.py#L2013) ? And append the fstype property to r ?

As the serialization code doesn't look at underscore attributes, or the property apparently, I'm suggesting that we revert fstype from a property like you had originally suggested - but fix the code that breaks on the ZFS codepath.

Reply:
> Try adding something like this:

This is now added on 5d602a3080d1f4ce4c79008cf865133be750ede2

As for the ZFS unit test case, I'm not sure how to proceed; if you can point me to a similar test already done or some pointers, I would be grateful.